gtsfm/utils/metrics.py

```diff
@@ -167,19 +167,71 @@ def compute_averaging_metrics(
         else:
             wTi_list.append(Pose3(wRi, wti))
 
-    # ground truth is the reference/target for alignment
-    wTi_aligned_list = comp_utils.align_poses_sim3_ignore_missing(gt_wTi_list, wTi_list)
+    # ground truth is the reference/target for alignment. discard 2nd return arg -- the estimated Similarity(3) object
+    wTi_aligned_list, _ = comp_utils.align_poses_sim3_ignore_missing(gt_wTi_list, wTi_list)
+
+    i2Ui1_dict_gt = simulate_twoview_translation_direction_measurements(gt_wTi_list)
 
     wRi_aligned_list, wti_aligned_list = get_rotations_translations_from_poses(wTi_aligned_list)
     gt_wRi_list, gt_wti_list = get_rotations_translations_from_poses(gt_wTi_list)
 
     metrics = []
     metrics.append(compute_rotation_angle_metric(wRi_aligned_list, gt_wRi_list))
     metrics.append(compute_translation_distance_metric(wti_aligned_list, gt_wti_list))
-    metrics.append(compute_translation_angle_metric(i2Ui1_dict, wTi_aligned_list))
+    metrics.append(compute_translation_angle_metric(i2Ui1_dict=i2Ui1_dict_gt, wTi_list=wTi_aligned_list))
     return GtsfmMetricsGroup(name="averaging_metrics", metrics=metrics)
 
 
+def compute_ba_pose_metrics(
```

Comment:
the main issue is that otherwise we would need to pass the GT pose graph into the BA code, and do the alignment there. That seems like a completely separate operation. So the other option would be to append to the GtsfmMetricsGroup for BA, after it's been created? Does that seem better?

Reply:
I think all BA metrics should be computed in one place, makes code easier to understand for someone who did not write it. 
Ideally we would have to pass the GT poses to bundle adjustment, but that also has its cons API wise. The multi-view optimizer has the GT poses and some BA metrics, maybe we can compute these metrics there and add to the BA metrics group there.

Reply:
Agreed. But I feel like aligning two scenes is a scene-specific operation, so I feel like it belongs in the SceneOptimizer.

Reply:
we are aligning poses for BA metrics, I think that is a detail people reading scene optimizer code generally would not care about.

Reply:
and you could also add it to the BA metrics group there. so scene optimizer just receives the metrics it cares about (for saving them).

Reply:
Agree with Akshay that metrics of one module should be populated in one place. We do pass in GT for other modules (e.g. two view) to generate metrics, so we can do that I think.

Maybe later we can come back and refactor code.